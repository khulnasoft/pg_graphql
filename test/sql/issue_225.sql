begin;
    -- https://github.com/khulnasoft/pg_graphql/issues/225

    create table post(
        id int primary key,
        title text
    );

    insert into public.post(id, title)
    select x.id, (10-x.id)::text from generate_series(1,3) x(id);

    select jsonb_pretty(
        graphql.resolve($$
            {
              postCollection( orderBy: [{id: DescNullsFirst, title: null}]) {
                edges {
                  node {
                    id
                    title
                  }
                }
              }
            }
        $$)
    );

    select jsonb_pretty(
        graphql.resolve($$
            {
              postCollection( orderBy: [{id: null, title: DescNullsLast}]) {
                edges {
                  node {
                    id
                    title
                  }
                }
              }
            }
        $$)
    );

    select jsonb_pretty(
        graphql.resolve($$
            {
              postCollection( orderBy: [{id: null}, { title: DescNullsLast}]) {
                edges {
                  node {
                    id
                    title
                  }
                }
              }
            }
        $$)
    );


rollback;
