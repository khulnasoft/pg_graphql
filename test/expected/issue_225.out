begin;
    -- https://github.com/khulnasoft/pg_graphql/issues/225
    create table post(
        id int primary key,
        title text
    );
    insert into public.post(id, title)
    select x.id, (10-x.id)::text from generate_series(1,3) x(id);
    select jsonb_pretty(
        graphql.resolve($$
            {
              postCollection( orderBy: [{id: DescNullsFirst, title: null}]) {
                edges {
                  node {
                    id
                    title
                  }
                }
              }
            }
        $$)
    );
             jsonb_pretty             
--------------------------------------
 {                                   +
     "data": {                       +
         "postCollection": {         +
             "edges": [              +
                 {                   +
                     "node": {       +
                         "id": 3,    +
                         "title": "7"+
                     }               +
                 },                  +
                 {                   +
                     "node": {       +
                         "id": 2,    +
                         "title": "8"+
                     }               +
                 },                  +
                 {                   +
                     "node": {       +
                         "id": 1,    +
                         "title": "9"+
                     }               +
                 }                   +
             ]                       +
         }                           +
     }                               +
 }
(1 row)

    select jsonb_pretty(
        graphql.resolve($$
            {
              postCollection( orderBy: [{id: null, title: DescNullsLast}]) {
                edges {
                  node {
                    id
                    title
                  }
                }
              }
            }
        $$)
    );
             jsonb_pretty             
--------------------------------------
 {                                   +
     "data": {                       +
         "postCollection": {         +
             "edges": [              +
                 {                   +
                     "node": {       +
                         "id": 1,    +
                         "title": "9"+
                     }               +
                 },                  +
                 {                   +
                     "node": {       +
                         "id": 2,    +
                         "title": "8"+
                     }               +
                 },                  +
                 {                   +
                     "node": {       +
                         "id": 3,    +
                         "title": "7"+
                     }               +
                 }                   +
             ]                       +
         }                           +
     }                               +
 }
(1 row)

    select jsonb_pretty(
        graphql.resolve($$
            {
              postCollection( orderBy: [{id: null}, { title: DescNullsLast}]) {
                edges {
                  node {
                    id
                    title
                  }
                }
              }
            }
        $$)
    );
             jsonb_pretty             
--------------------------------------
 {                                   +
     "data": {                       +
         "postCollection": {         +
             "edges": [              +
                 {                   +
                     "node": {       +
                         "id": 1,    +
                         "title": "9"+
                     }               +
                 },                  +
                 {                   +
                     "node": {       +
                         "id": 2,    +
                         "title": "8"+
                     }               +
                 },                  +
                 {                   +
                     "node": {       +
                         "id": 3,    +
                         "title": "7"+
                     }               +
                 }                   +
             ]                       +
         }                           +
     }                               +
 }
(1 row)

rollback;
